pkg/proc/mem.go

```diff
@@ -129,8 +130,41 @@ func (mem *compositeMemory) ReadMemory(data []byte, addr uint64) (int, error) {
 }
 
 func (mem *compositeMemory) WriteMemory(addr uint64, data []byte) (int, error) {
-	//TODO(aarzilli): implement
-	return 0, errors.New("can't write composite memory")
+	addr -= fakeAddress
+	if addr >= uint64(len(mem.data)) || addr+uint64(len(data)) > uint64(len(mem.data)) {
+		return 0, errors.New("write out of bounds")
+	}
+	if mem.regs.ChangeFunc == nil {
+		return 0, errors.New("can not write registers")
```

Comment:
Will this error ever propagate to the user? If so we should make it a little friendlier and explanatory.

Reply:
Additionally, does the absence of a `ChangeFunc` indicate that register writing is not supported by the current debug mode, e.g. debugging core files? Or should we panic here to indicate a programmer error that we never properly set this value at startup in some area of the code.